lib/widgets/message_list.dart

```diff
@@ -523,11 +532,58 @@ class MarkAsReadWidget extends StatelessWidget {
                     height: (23 / 18))
                   .merge(weightVariableTextStyle(context, wght: 400))),
                 shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(7)),
+              ).copyWith(
+                // clobber `FilledButton`'s default disabled state
+                // TODO(#95) need dark-theme colors (foreground and background)
+                foregroundColor: WidgetStateColor.resolveWith((_) => Colors.white),
+                backgroundColor: WidgetStateColor.resolveWith((_) => _UnreadMarker.color),
```

Comment:
Ah:

Wo the function you pass to [`WidgetStateColor.resolveWith`](https://api.flutter.dev/flutter/widgets/WidgetStateColor/resolveWith.html) receives a set of [`WidgetState`](https://api.flutter.dev/flutter/widgets/WidgetState.html)s, which are hovered, focused, pressed, etc., and disabled. Normally, you'd pass a function that checks that set of states and returns different colors based on it.

Here, we don't want different colors for the different states. We don't even want a different color for the disabled state; that's because we have other code for that. That other code uses an animation, which is especially helpful at the end of the loading state, when otherwise the button would jump to full opacity before the mark-as-read event arrives and the exit animation starts. And it doesn't seem possible to animate between disabled/enabled here using `ButtonStyle`.

Reply:
(Oh and the other key piece of information: without passing these arguments, where we set the colors unconditionally, the widget would choose its own colors for the disabled state.)

Reply:
Cool, thanks. @Khader-1 here's a way to explain that in the comment:
```suggestion
              ).copyWith(
                // Give the buttons a constant color regardless of whether their
                // state is disabled, pressed, etc.  We handle those states
                // separately, via MarkAsReadAnimation.
                // TODO(#95) need dark-theme colors (foreground and background)
                foregroundColor: WidgetStateColor.resolveWith((_) => Colors.white),
                backgroundColor: WidgetStateColor.resolveWith((_) => _UnreadMarker.color),
```